src/main/java/org/truffleruby/core/numeric/IntegerNodes.java

```diff
@@ -469,9 +575,42 @@ public Object mod(long a, DynamicObject b) {
             return mod;
         }
 
+        @TruffleBoundary
+        @Specialization
+        public Object mod(DynamicObject a, long b) {
+            if (b == 0) {
+                throw new ArithmeticException("divide by zero");
```

Comment:
I know you just moved this code, but shouldn't this be a Ruby exception?

Reply:
It's translated by ExceptionTranslationNode.